src/membership/schemas/membership.schema.ts

```diff
@@ -0,0 +1,50 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+import { MembershipStatus } from '../enums/membership-status.enum';
+import { IsEnum } from 'class-validator';
+
+@Schema({ collection: 'memberships' })
+export class Membership extends Document {
+  @Prop({ required: true, minlength: 5, maxlength: 50 })
+  name: string;
+
+  @Prop({ required: true, unique: true })
+  mobileNo: string;
+
+  @Prop({ required: true, unique: true })
+  email: string;
+
+  @Prop({ required: true })
+  currentCountry: string;
+
+  @Prop()
+  profession: string;
+
+  @Prop()
+  institute: string;
+
+  @Prop()
+  designation: string;
+
+  @Prop()
+  lastAcademicLevel: string;
+
+  @Prop()
+  lastPassingYear: number;
+
+  @Prop()
+  lastBatch: string;
```

Comment:
these should be entityRef. Means, these property should refer related data. for example -

```suggestion
  currentCountry: {
      id: "",
      name: ""
  }
```

Talk with me if you dont understand.